cli/main.rs

```diff
@@ -540,9 +541,36 @@ async fn run_repl(flags: Flags) -> Result<(), ErrBox> {
 
 async fn run_command(flags: Flags, script: String) -> Result<(), ErrBox> {
   let global_state = GlobalState::new(flags.clone())?;
-  let main_module = ModuleSpecifier::resolve_url_or_path(&script).unwrap();
+  let main_module = if script != "-" {
+    ModuleSpecifier::resolve_url_or_path(&script).unwrap()
+  } else {
+    ModuleSpecifier::resolve_url_or_path("./__$deno$stdin.ts").unwrap()
+  };
   let mut worker =
     MainWorker::create(global_state.clone(), main_module.clone())?;
+  if script == "-" {
+    let mut source = Vec::new();
+    std::io::stdin().read_to_end(&mut source)?;
+    let main_module_url = main_module.as_url().to_owned();
+    // Create a dummy source file.
+    let source_file = SourceFile {
+      filename: main_module_url.to_file_path().unwrap(),
+      url: main_module_url,
+      types_url: None,
+      types_header: None,
+      media_type: MediaType::Unknown,
```

Comment:
`MediaType::Unknown` is treated as JS - if you pass some typescript code it won't work. It should probably be handled like in `deno eval` - by providing `--ts` flag to tell that it's TypeScript - but it would mean adding new flag for `deno run` which is not so great...

Reply:
Is there any negative about treating as any input as typescript? If there is a `--ts` flag it might be good to be consistent and use it for overwriting the media type for real files as well.

Reply:
> Is there any negative about treating as any input as typescript?

It will always spawn TS compiler and perform type check. This would be fixed by `--no-type-check` flag though... So I guess we can default to TS after all

Reply:
I dont see `--no-type-check` in the documentation right now, will this be added later?
I changed over to TS by default.

Reply:
> I dont see --no-type-check in the documentation right now, will this be added later?

Yes, more in the issue: #5436 

> I changed over to TS by default.

👍